### ISSUE TITLE
[FEEDBACK]: Invoice ID

### ISSUE THREAD

**Issue by narciso-cerezo** [FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec#607]
### Proposed Change

Assigning a proper **Invoice ID** to each cost line when the invoices are generated.
AWS already does this in the CUR.
This is necessary to properly match all usage against a given invoice and conciliate payments.

### What use cases, FinOps or others, can't be performed with the current specification without this change?

We need to distribute all cost back to each cost center and then group cost by **Billing Account ID**, **Invoice Issuer** and **Invoice ID** so that our finance department can charge back the business units and pay the invoices.
The sum of the invoices for a given Billing Account ID and Invoice Issuer must match to the cent the sum of the chargeback per business unit for that same perimeter.
Without Invoice ID there are situations like when an account is moved between different Billing Accounts, where you loose track and are unable to properly match those.

Another important use case are Credit Memos (Refund Invoices): AWS creates Credit Memos and they show up in the CUR at that point in time, however there is a manual process involved to request that Credit Memo to be applied to a given regular Invoice. The result is that you cannot usually include the credit memo in the same billing period. We set the applicable billing period for each Credit Memo to take them into account, and for that we need the Invoice ID to be able to discriminate the right rows of data in the input.

### Which FinOps personas perform this use case?

FinOps Practitioner, Finance

### For which providers do you perform this use case for?

AWS, Azure, GCP

### Criticality Scale

1: Blocks my organization from adopting FOCUS

### Context / Supporting information

We are using the CUR for chargeback and invoicing in AWS and the Invoice ID is mandatory for us.
We have also implemented the same process with Azure, and GCP is in the works, but since the Azure Cost Exports do not include the Invoice ID we need to apply heuristics, leading to errors some times and manual intervention.
Adding the Invoice ID in all three providers would help us implement the exact same process for all three, which would be great.

**Comment by shawnalpay** [comment]
@narciso-cerezo I'm happy to report that we already have an issue for this one!

- #302

That said, we are in the midst of revising our issue templates -- so stay tuned.  We may either close this issue or #302, but we may want to gather more information from you soon.  Thanks!

**Comment by jpradocueva** [comment]
### Summary TF-2 call Oct 16:
[#607](https://github.com/FinOps-Open-Cost-and-Usage-Spec/FOCUS_Spec/issues/607) Feedback on Invoice ID 
Primary Issue: A request for adding invoice ID, which already exists in another issue.
Core Problem: Redundancy between issues, indicating more clarity is needed in the backlog.
Divergent Views: None; the need for the feature was affirmed, but it was noted that the issue already exists.
Final Agreement: The issue will remain open until the issue template is revised, at which point it may be closed.
Action Items: 
None are assigned at this point.

**Comment by shawnalpay** [comment]
@narciso-cerezo We've now crafted #615, which is a `Work Item` for adding Invoice ID to the spec, and it's currently in the running for being added to 1.2!  I've added some of your quotes to that issue, and I've also linked this one there for full reference.  So I'm going to close this issue (and Michael's duplicate #302) in favor of that one, and we can discuss further there.  Thanks for filing this!
